snuba/migrations/runner.py

```diff
@@ -434,3 +434,69 @@ def add_node(
             elif isinstance(migration, CodeMigration):
                 for python_op in migration.forwards_global():
                     python_op.execute_new_node(storage_sets)
+
+    @classmethod
+    def copy_tables(
+        self,
+        node_type: ClickhouseNodeType,
+        host_name: str,
+        port: int,
+        user: str,
+        password: str,
+        database: str,
+        new_port: int,
+        table_names: Optional[str],
+        dry_run: bool = False,
+    ) -> None:
+        """
+        Create tables on a new node in a cluster by using the existing create table
+        statement from an existing node in a cluster. This can be done for a new shard
+        or new replica.
+        """
+        client_settings = ClickhouseClientSettings.QUERY.value
+        clickhouse = ClickhousePool(
+            host_name,
+            port,
+            user,
+            password,
+            database,
+            client_settings=client_settings.settings,
+            send_receive_timeout=client_settings.timeout,
+        )
+
+        # TODO: assert that the new node is not in the system.clusters definitions yet
+
+        if table_names:
+            ch_table_names = table_names.split(",")
+        else:
+            ch_table_names = [
+                result[0] for result in clickhouse.execute("SHOW TABLES").results
+            ]
```

Comment:
I think this is going to be a bit fragile. If we trust the content of the tables on another node we should at least validate they are the same you have in the storage set you are trying to deploy on this node. 
More in general, how do you expect this script to be used? Is the user going to tell "please duplicate this node" or is the user going to tell "Please create all the tables you should have on this node according to the storage sets on this cluster" ?